apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SpannerDate.cs

```diff
@@ -0,0 +1,408 @@
+        /// </summary>
+        /// <returns>The DateTime value.</returns>
+        public DateTime ToDateTime() => DateTime.MinValue.AddDays(_days);
+
+        /// <summary>
+        /// Converts the value of SpannerDate to The DateTimeOffset.
+        /// </summary>
+        /// <value>
+        /// To date time offset.
+        /// </value>
+        public DateTimeOffset ToDateTimeOffset => new DateTimeOffset(ToDateTime());
+
+        // DateTime Properties
+
+        /// <summary>
+        /// Gets the day component of the date represented by this instance.
+        /// </summary>
+        /// <value>
+        /// The day.
+        /// </value>
+        public int Day => ToDateTime().Day;
+
+        /// <summary>
+        /// Gets the month component of the date represented by this instance.
+        /// </summary>
+        /// <value>
+        /// The month.
+        /// </value>
+        public int Month => ToDateTime().Month;
+
+        /// <summary>
+        /// Gets the year component of the date represented by this instance.
+        /// </summary>
+        /// <value>
+        /// The year.
+        /// </value>
+        public int Year => ToDateTime().Year;
+
+        /// <summary>
+        /// Gets the day of year represented by this instance.
+        /// </summary>
+        /// <value>
+        /// The day of year.
+        /// </value>
+        public int DayOfYear => ToDateTime().DayOfYear;
+
+        /// <summary>
+        /// Gets the day of week represented by this instance.
+        /// </summary>
+        /// <value>
+        /// The day of week.
+        /// </value>
+        public DayOfWeek DayOfWeek => ToDateTime().DayOfWeek;
+
+        /// <summary>
+        /// Gets the number of days since January 1, 0001 represented by this instance.
+        /// </summary>
+        /// <value>
+        /// The day number.
+        /// </value>
+        public int DayNumber => _days;
+
+        // Operators
+
+        /// <summary>
+        /// Compares two values for equality.
+        /// </summary>
+        /// <param name="lhs">The first value to compare.</param>
+        /// <param name="rhs">The second value to compare.</param>
+        /// <returns><c>true</c> if the two arguments are equal; <c>false</c> otherwise.</returns>
+        public static bool operator ==(SpannerDate lhs, SpannerDate rhs) => lhs.Equals(rhs);
+
+        /// <summary>
+        /// Compares two values for inequality.
+        /// </summary>
+        /// <param name="lhs">The first value to compare.</param>
+        /// <param name="rhs">The second value to compare.</param>
+        /// <returns><c>false</c> if the two arguments are equal; <c>true</c> otherwise.</returns>
+        public static bool operator !=(SpannerDate lhs, SpannerDate rhs) => !lhs.Equals(rhs);
+
+        /// <summary>Compares two values to determine which is less.</summary>
+        /// <param name="lhs">The value to compare with <paramref name="lhs" />.</param>
```

Comment:
The paramrefs here and below are incorrect - they're all saying they compare with the same parameter they're defining.